lib/Conversion/LLHDToLLVM/LLHDToLLVM.cpp

```diff
@@ -1132,19 +1786,19 @@ struct ShrOpConversion : public ConvertToLLVMPattern {
 
     if (auto resTy = shrOp.result().getType().dyn_cast<IntegerType>()) {
       // Get width of the base and hidden values combined.
-      auto baseWidth = shrOp.getType().getIntOrFloatBitWidth();
-      auto hdnWidth = shrOp.hidden().getType().getIntOrFloatBitWidth();
+      auto baseWidth = getStdOrLLVMIntegerWidth(shrOp.getType());
+      auto hdnWidth = getStdOrLLVMIntegerWidth(shrOp.hidden().getType());
       auto full = baseWidth + hdnWidth;
 
```

Comment:
Looking forward to finally get rid of these weird shift ops in the LLHD dialect soonish. Pretty bad design choice on my part back in the day 😇 . But that's for a separate PR later.